ui/src/app/views/notebook-redirect/component.ts

```diff
@@ -30,6 +30,25 @@ enum Progress {
   Redirecting
 }
 
+export enum Kernels {
+  R,
+  Python3,
+}
+
+const rNotebookFileContent = '{ "cells": [ { "cell_type": "code", "execution_count": null, \
```

Comment:
Another idea which might be viable but which I have not thoroughly investigated is whether we can keep the creation logic the same, but modify the redirection URI to specify a different kernel type. On first save, this would stick.

Things to check on that approach:
- Whether a newly created notebook currently has a type and kernel type, or it is typeless and the Jupyter server is picking  adefault kernel
- Whether we might need to determine this information immediately after creation in order to display the notebook type on the list page (this might be an argument to continue with your current approach). 

Actually trying this out on Jupyter, I'm not sure there's a way to encode this in the URL so it may be a dead-end.

Reply:
Part of the goal here is also to do a majority of the work through our UI, rather than Jupyter's, because it means things get preserved in our cache.